client/reader/sidebar/index.jsx

```diff
@@ -254,22 +235,51 @@ const ReaderSidebar = React.createClass( {
 					onTagExists={ this.highlightNewTag }
 					currentTag={ this.state.currentTag } />
 
-				</SidebarRegion>
+			</SidebarRegion>
 
-				{ this.renderAppPromo() }
+			{ this.props.shouldRenderAppPromo &&
+				<div className="sidebar__app-promo">
+					<AppPromo location="reader" locale={ userUtils.getLocaleSlug() } />
+				</div>
+			}
 
 				<SidebarFooter />
 			</Sidebar>
 		);
 	}
 } );
 
+ReaderSidebar.defaultProps = {
+	translate: identity
+};
+
+export const shouldRenderAppPromo = ( options = { } ) => {
+	const {
+		isDesktopPromoDisabled = store.get( 'desktop_promo_disabled' ),
+		isViewportMobile = viewport.isMobile(),
+		isUserLocaleEnglish = 'en' === userUtils.getLocaleSlug(),
+		isDesktopPromoConfiguredToRun = config.isEnabled( 'desktop-promo' ),
+		isUserDesktopAppUser = userSettings.getSetting( 'is_desktop_app_user' ),
+		isUserOnChromeOs = /\bCrOS\b/.test( navigator.userAgent )
+	} = options;
+
+	return every( [
+		! isDesktopPromoDisabled,
+		isUserLocaleEnglish,
+		! isViewportMobile,
+		! isUserOnChromeOs,
+		isDesktopPromoConfiguredToRun,
+		! isUserDesktopAppUser
+	] );
+};
+
 export default connect(
 	( state ) => {
 		return {
 			isListsOpen: state.ui.reader.sidebar.isListsOpen,
 			isTagsOpen: state.ui.reader.sidebar.isTagsOpen,
-			subscribedLists: getSubscribedLists( state )
+			subscribedLists: getSubscribedLists( state ),
+			shouldRenderAppPromo: shouldRenderAppPromo(),
```

Comment:
@dmsnell : I'm not sure I can agree on that. The purpose of `connect` as far as I can see it is to provide props to the component as made available via the `context` store instance, of which `sholudRenderAppPromo` does not use.